docs/guides/scheduler_event.md

```diff
@@ -0,0 +1,107 @@
+
+Take an example of parsing a `message create` event using scheduler:
+
+```c
+enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
+    if (event == DISCORD_EV_MESSAGE_CREATE) {
+        struct discord_message self;
+        /* Parse the JSON and store the it in the "self" struct. */
+        discord_message_from_json(data, length, &self);
+
+        log_debug("Message content: %s", self.content);
+
+        /* Free the struct. */
+        discord_message_cleanup(&self);
+        return DISCORD_EVENT_IGNORE;
+    }
+    return DISCORD_EVENT_MAIN_THREAD;
+}
+```
+
+Mhm, that's easy! Now, if you want to parse a single JSON key, you should see the next section.
+
+## How to parse the JSON (jsmnf)
+
+This method is a bit more complicated, but it is the fastest one, and you will not need to free the returned struct, so it's a win-win situation.
+
+Note that the `jsmn-find` library, the one we are going to use, it's the same which Concord uses for parsing.
+
+Take an example of parsing a message content from `message create` event with scheduler:
+
+```c
+enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
+    if (event == DISCORD_EV_MESSAGE_CREATE) {
+        jsmn_parser parser;
+        jsmntok_t tokens[1024];
+
+        /* Set base variables to the parser struct. */
+        jsmn_init(&parser);
+        /* Parses the JSON and store the tokens. */
+        int r = jsmn_parse(&parser, data, length, tokens, sizeof(tokens));
+
+        /* Check if the parsing was successful. */
+        if (r < 0) {
+            log_error("Failed to parse JSON.");
+            return DISCORD_EVENT_IGNORE;
+        }
+
+        jsmnf_loader loader;
+        jsmnf_pair pairs[1024];
+
+        /* Set base variables of the loader struct. */
+        jsmnf_init(&loader);
+        /* Load jsmn-find with the tokens and the pairs. */
+        r = jsmnf_load(&loader, data, tokens, parser.toknext, pairs, sizeof(pairs) / sizeof(*pairs));
+
+        if (r < 0) {
+            log_error("Failed to load jsmn-find.");
+            return DISCORD_EVENT_IGNORE;
+        }
+
+        jsmnf_pair *content = jsmnf_find(pairs, data, "content", strlen("content"));
+        if (!content) {
+            log_error("Failed to find the \"content\" key.");
+            return DISCORD_EVENT_IGNORE;
+        }
+
+        /* The maximum length of a content in Discord is 2000, so we will be used it */
+        char contentStr[2000 + 1];
+        snprintf(contentStr, sizeof(contentStr), "%.*s", (int)content->v.len, data + content->v.pos);
+
+        log_debug("Message content: %s", contentStr);
+        return DISCORD_EVENT_IGNORE;
+    }
+    return DISCORD_EVENT_MAIN_THREAD;
+}
+```
+
+Done! You will be parsing only the `content` key, and this allows you to build more features and systems, and use all that Concord has to offer.
```

Comment:
```suggestion
This will only parse the `content` key allows you to build more features and systems.
```
Like before, "Done!" and "use all that Concord has to offer" doesn't contribute anything to the guide and might be a touch misleading. I suggest keeping out any potential results that aren't limited to the guide.

Reply:
Did you mean to change "allows" to "allowing"? And "Done! ..." doesn't seems to be so bad at all, as it was included on the sqlite3 guide for example, but in case you want, I can do it.